src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,32 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
+import {LoadDirection} from './enum/load-direction';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    @Output()
+    readonly loadData = new EventEmitter<LoadDirection>();
+
+    private currentScrollTop = 0;
+
+    @HostListener('scroll', ['$event.target'])
+    protected onScroll(target: HTMLElement): void {
+        const {scrollHeight, scrollTop, clientHeight} = target;
```

Comment:
Я лично пропагандирую такой подход, он на практике хорошо зарекомендовал себя:
```suggestion
    protected onScroll({scrollHeight, scrollTop, clientHeight}: HTMLElement): void {
```

А почему метод `protected`?)

Reply:
Исправила, заменила на private